Private Profile Photos

Samyoul opened the pull request:
## What has changed?

**Settings**

I've added a new setting `profilePicturesShowTo` which has three options for whom should see the user's profile picture:

- `ProfilePicturesShowToContactsOnly` - shows only to the user's contacts *default*
- `ProfilePicturesShowToEveryone` - shows their profile image to everyone
- `ProfilePicturesShowToNone` - shows their profile image to no-one, even if an image is set.

I've also added functionality that does the following:

**On Sending**

- If setting `profilePicturesShowTo` is set to `ProfilePicturesShowToContactsOnly`
- Create an AES encryption key and encrypt all the user's identity image payloads
- Replace the plain payload with the encrypted payload
- Encrypt the encryption AES key using ECDH AES encryption for each of the user's contacts
- Attach all the encrypted encryption AES key payloads to the `IdentityImage` protobuf
- Send the protobuf as usual

**On Receiving**

- For every `ChatIdentity` that is received attempt to decrypt any attached `IdentityImages`
  - loop through all encrypted encryption AES key payloads
  - If a encrypted encryption AES key payload can be decrypted using an ECDH key between sender and recipient
  - attempt to decrypt the encrypted image payload
  - on success replace the encrypted payload with the plain decrypted payload
- After the decryption process any images that remain encrypted are discarded and not processed any further
- Otherwise process the images as usual

## Why make the change?

In line with offering a privacy first product images and the concerns raised in https://github.com/status-im/status/issues/56 this change was introduced to ensure a user could use profile images without sharing their image with everyone they interacted with on Status if they so choose not to.

Samyoul committed:
Added encryption keys to the IdentityImage protobuf

Samyoul committed:
Added getSettings helper and moved IdentityImage attachment to dedicated function

Samyoul committed:
Added encryption element to

Samyoul committed:
Switched encryption of payload and keys to AES

Added Encrypted field to the protobuf

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [x] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (178)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 033e76fa | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/1/) | 2021-02-12 13:10:26 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210212-130804-033e76.zip) |
| :heavy_check_mark: | 033e76fa | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/1/) | 2021-02-12 13:12:14 | ~4 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210212-130803-033e76.zip) |
| :heavy_check_mark: | 033e76fa | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/1/) | 2021-02-12 13:14:11 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210212-130800-033e76.aar) |
| | | | | | | |
| :heavy_check_mark: | d5be3c5c | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/2/) | 2021-02-12 15:38:45 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210212-153608-d5be3c.zip) |
| :heavy_check_mark: | d5be3c5c | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/2/) | 2021-02-12 15:38:45 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210212-153607-d5be3c.zip) |
| :heavy_check_mark: | d5be3c5c | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/2/) | 2021-02-12 15:43:51 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210212-153603-d5be3c.aar) |
| | | | | | | |
| :heavy_check_mark: | c1ccb8f3 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/3/) | 2021-02-15 17:45:16 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210215-174312-c1ccb8.zip) |
| :heavy_check_mark: | c1ccb8f3 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/3/) | 2021-02-15 17:45:42 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210215-174312-c1ccb8.zip) |
| :heavy_check_mark: | c1ccb8f3 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/3/) | 2021-02-15 17:48:27 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210215-174312-c1ccb8.aar) |
| | | | | | | |
| :heavy_check_mark: | 4d9fe726 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/4/) | 2021-02-16 12:18:54 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210216-121609-4d9fe7.zip) |
| :heavy_check_mark: | 4d9fe726 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/4/) | 2021-02-16 12:21:34 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210216-121507-4d9fe7.aar) |
| :heavy_check_mark: | 4d9fe726 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/4/) | 2021-02-16 12:23:12 | ~7 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210216-121610-4d9fe7.zip) |
| | | | | | | |
| :heavy_check_mark: | 0c68a8cb | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/5/) | 2021-02-16 16:16:07 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210216-161349-0c68a8.zip) |
| :heavy_check_mark: | 0c68a8cb | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/5/) | 2021-02-16 16:18:22 | ~4 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210216-161349-0c68a8.zip) |
| :heavy_check_mark: | 0c68a8cb | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/5/) | 2021-02-16 16:20:21 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210216-161349-0c68a8.aar) |
| | | | | | | |
| :heavy_check_mark: | e83c4df5 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/6/) | 2021-02-17 23:16:51 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210217-231518-e83c4d.zip) |
| :heavy_check_mark: | e83c4df5 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/6/) | 2021-02-17 23:16:57 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210217-231518-e83c4d.zip) |
| :heavy_check_mark: | e83c4df5 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/6/) | 2021-02-17 23:21:10 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210217-231518-e83c4d.aar) |
| | | | | | | |
| :heavy_check_mark: | d0c986cd | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/7/) | 2021-02-18 14:38:32 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210218-143633-d0c986.zip) |
| :heavy_check_mark: | d0c986cd | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/7/) | 2021-02-18 14:39:07 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210218-143633-d0c986.zip) |
| :heavy_check_mark: | d0c986cd | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/7/) | 2021-02-18 14:41:27 | ~4 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210218-143633-d0c986.aar) |
| | | | | | | |
| :heavy_check_mark: | 69541c5d | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/8/) | 2021-02-18 14:43:53 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210218-144223-69541c.zip) |
| :heavy_check_mark: | 69541c5d | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/8/) | 2021-02-18 14:43:56 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210218-144225-69541c.zip) |
| :heavy_check_mark: | 69541c5d | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/8/) | 2021-02-18 14:47:06 | ~4 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210218-144223-69541c.aar) |
| | | | | | | |
| :heavy_check_mark: | dbf1d88f | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/9/) | 2021-02-18 14:49:57 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210218-144755-dbf1d8.zip) |
| :heavy_check_mark: | dbf1d88f | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/9/) | 2021-02-18 14:49:59 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210218-144755-dbf1d8.zip) |
| :heavy_check_mark: | dbf1d88f | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/9/) | 2021-02-18 14:53:53 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210218-144753-dbf1d8.aar) |
| | | | | | | |
| :heavy_check_mark: | 7b0751cd | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/10/) | 2021-02-18 15:16:51 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210218-151524-7b0751.zip) |
| :heavy_check_mark: | 7b0751cd | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/10/) | 2021-02-18 15:17:51 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210218-151524-7b0751.zip) |
| :heavy_check_mark: | 7b0751cd | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/10/) | 2021-02-18 15:21:35 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210218-151524-7b0751.aar) |
| | | | | | | |
| :heavy_check_mark: | a748542b | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/11/) | 2021-02-18 15:28:52 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210218-152720-a74854.zip) |
| :heavy_check_mark: | a748542b | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/11/) | 2021-02-18 15:29:02 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210218-152721-a74854.zip) |
| :heavy_check_mark: | a748542b | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/11/) | 2021-02-18 15:32:24 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210218-152719-a74854.aar) |
| | | | | | | |
| :heavy_check_mark: | 0328607b | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/12/) | 2021-02-18 15:46:47 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210218-154504-032860.zip) |
| :heavy_check_mark: | 0328607b | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/12/) | 2021-02-18 15:46:59 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210218-154504-032860.zip) |
| :heavy_check_mark: | 0328607b | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/12/) | 2021-02-18 15:51:07 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210218-154504-032860.aar) |
| | | | | | | |
| :heavy_check_mark: | 5f2a731e | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/13/) | 2021-02-18 15:48:17 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210218-154649-5f2a73.zip) |
| :heavy_check_mark: | 5f2a731e | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/13/) | 2021-02-18 15:48:48 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210218-154701-5f2a73.zip) |
| :heavy_check_mark: | 5f2a731e | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/13/) | 2021-02-18 15:56:38 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210218-155109-5f2a73.aar) |
| | | | | | | |
| :heavy_check_mark: | ff8d248f | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/14/) | 2021-02-19 17:00:19 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210219-165819-ff8d24.zip) |
| :heavy_check_mark: | ff8d248f | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/14/) | 2021-02-19 17:00:42 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210219-165816-ff8d24.zip) |
| :heavy_check_mark: | ff8d248f | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/14/) | 2021-02-19 17:03:35 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210219-165816-ff8d24.aar) |
| | | | | | | |
| :heavy_check_mark: | 28a87ec7 | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/15/) | 2021-02-26 15:11:59 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210226-151008-28a87e.zip) |
| :heavy_check_mark: | 28a87ec7 | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/15/) | 2021-02-26 15:13:48 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210226-151008-28a87e.zip) |
| :heavy_check_mark: | 28a87ec7 | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/15/) | 2021-02-26 15:15:33 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210226-151003-28a87e.aar) |
| | | | | | | |
| :heavy_check_mark: | bb9f1c22 | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/16/) | 2021-03-01 17:27:39 | ~12 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210301-171500-bb9f1c.aar) |
| :heavy_check_mark: | bb9f1c22 | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/16/) | 2021-03-01 17:28:39 | ~13 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210301-171503-bb9f1c.zip) |
| :heavy_check_mark: | bb9f1c22 | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/16/) | 2021-03-01 17:57:23 | ~42 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210301-171500-bb9f1c.zip) |
| | | | | | | |
| :x: | 319dfce2 | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/17/) | 2021-03-04 16:24:42 | ~49 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/17/consoleText) |
| :heavy_check_mark: | 319dfce2 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/17/) | 2021-03-04 16:25:58 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210304-162352-319dfc.zip) |
| :heavy_check_mark: | 319dfce2 | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/17/) | 2021-03-04 16:28:48 | ~4 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210304-162352-319dfc.zip) |
| :x: | 319dfce2 | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/18/) | 2021-03-04 16:44:22 | ~34 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/18/consoleText) |
| :x: | 319dfce2 | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/19/) | 2021-03-04 17:01:18 | ~11 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/19/consoleText) |
| :x: | 319dfce2 | [#20](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/20/) | 2021-03-05 08:03:39 | ~12 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/20/consoleText) |
| :x: | 319dfce2 | [#21](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/21/) | 2021-03-05 08:05:20 | ~7.2 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/21/consoleText) |
| :x: | 319dfce2 | [#22](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/22/) | 2021-03-05 08:11:46 | ~4.5 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/22/consoleText) |
| :x: | 319dfce2 | [#23](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/23/) | 2021-03-05 08:33:43 | ~4 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/23/consoleText) |
| :x: | 319dfce2 | [#24](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/24/) | 2021-03-05 08:43:31 | ~6 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/24/consoleText) |
| :heavy_check_mark: | 319dfce2 | [#25](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/25/) | 2021-03-05 08:56:05 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210305-085035-319dfc.aar) |
| | | | | | | |
| :heavy_check_mark: | a3c27f28 | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/18/) | 2021-03-21 13:36:18 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210321-133416-a3c27f.zip) |
| :heavy_check_mark: | a3c27f28 | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/18/) | 2021-03-21 13:39:34 | ~5 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210321-133416-a3c27f.zip) |
| :heavy_check_mark: | a3c27f28 | [#26](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/26/) | 2021-03-21 13:40:46 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210321-133416-a3c27f.aar) |
| | | | | | | |
| :heavy_check_mark: | f0ba3c23 | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/19/) | 2021-03-22 13:48:05 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210322-134552-f0ba3c.zip) |
| :heavy_check_mark: | f0ba3c23 | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/19/) | 2021-03-22 13:49:54 | ~4 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210322-134548-f0ba3c.zip) |
| :heavy_check_mark: | f0ba3c23 | [#27](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/27/) | 2021-03-22 13:51:49 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210322-134548-f0ba3c.aar) |
| | | | | | | |
| :heavy_check_mark: | 2422883f | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/20/) | 2021-03-31 10:59:47 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210331-105752-242288.zip) |
| :heavy_check_mark: | 2422883f | [#20](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/20/) | 2021-03-31 11:01:30 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210331-105752-242288.zip) |
| :heavy_check_mark: | 2422883f | [#28](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/28/) | 2021-03-31 11:04:09 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210331-105749-242288.aar) |
| | | | | | | |
| :x: | cfe074d4 | [#29](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/29/) | 2021-04-19 13:02:06 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/29/consoleText) |
| :x: | cfe074d4 | [#21](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/21/) | 2021-04-19 13:03:21 | ~3 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/21/consoleText) |
| :x: | cfe074d4 | [#21](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/21/) | 2021-04-19 13:03:28 | ~3 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/21/consoleText) |
| | | | | | | |
| :heavy_check_mark: | 9db1b50b | [#22](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/22/) | 2021-04-19 15:59:10 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210419-155624-9db1b5.zip) |
| :heavy_check_mark: | 9db1b50b | [#22](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/22/) | 2021-04-19 15:59:25 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210419-155624-9db1b5.zip) |
| :heavy_check_mark: | 9db1b50b | [#30](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/30/) | 2021-04-19 16:01:46 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210419-155624-9db1b5.aar) |
| | | | | | | |
| :heavy_check_mark: | 0b8c8e86 | [#23](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/23/) | 2021-04-19 16:04:54 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210419-160301-0b8c8e.zip) |
| :heavy_check_mark: | 0b8c8e86 | [#23](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/23/) | 2021-04-19 16:06:37 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210419-160301-0b8c8e.zip) |
| :heavy_check_mark: | 0b8c8e86 | [#31](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/31/) | 2021-04-19 16:09:27 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210419-160259-0b8c8e.aar) |
| | | | | | | |
| :heavy_check_mark: | 3e249cd8 | [#24](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/24/) | 2021-04-20 10:01:21 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210420-095929-3e249c.zip) |
| :heavy_check_mark: | 3e249cd8 | [#24](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/24/) | 2021-04-20 10:01:59 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210420-095930-3e249c.zip) |
| :heavy_check_mark: | 3e249cd8 | [#32](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/32/) | 2021-04-20 10:04:00 | ~4 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210420-095927-3e249c.aar) |
| | | | | | | |
| :heavy_check_mark: | 45316ef6 | [#25](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/25/) | 2021-04-27 15:27:53 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210427-152506-45316e.zip) |
| :heavy_check_mark: | 45316ef6 | [#33](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/33/) | 2021-04-27 15:32:21 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210427-152506-45316e.aar) |
| | | | | | | |
| :heavy_check_mark: | 99dff9c1 | [#26](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/26/) | 2021-04-27 15:49:43 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-210427-154716-99dff9.zip) |
| :heavy_check_mark: | 99dff9c1 | [#26](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/26/) | 2021-04-27 15:51:50 | ~4 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-210427-154716-99dff9.zip) |
| :heavy_check_mark: | 99dff9c1 | [#34](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/34/) | 2021-04-27 15:53:45 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-210427-154716-99dff9.aar) |
| | | | | | | |
| :x: | 28b6b933 | [#35](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/35/) | 2021-06-07 14:41:47 | ~10 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/35/consoleText) |
| :x: | 28b6b933 | [#27](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/27/) | 2021-06-07 14:41:56 | ~10 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/27/consoleText) |
| :heavy_check_mark: | 28b6b933 | [#27](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/27/) | 2021-06-07 14:44:03 | ~13 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210607-143102-28b6b9.zip) |
| | | | | | | |
| :x: | e67a6da9 | [#36](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/36/) | 2021-08-09 12:46:24 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/36/consoleText) |
| :x: | e67a6da9 | [#28](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/28/) | 2021-08-09 12:47:20 | ~2 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/28/consoleText) |
| :heavy_check_mark: | e67a6da9 | [#28](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/28/) | 2021-08-09 12:48:00 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210809-124521-e67a6d.zip) |
| | | | | | | |
| :heavy_check_mark: | 236acd2d | [#29](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/29/) | 2021-08-10 12:56:05 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210810-125225-236acd.zip) |
| :heavy_check_mark: | 236acd2d | [#29](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/29/) | 2021-08-10 12:56:21 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210810-125225-236acd.zip) |
| :heavy_check_mark: | 236acd2d | [#37](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/37/) | 2021-08-10 12:58:57 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210810-125220-236acd.aar) |
| | | | | | | |
| :heavy_check_mark: | a19edfc6 | [#30](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/30/) | 2021-08-10 13:30:05 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210810-132747-a19edf.zip) |
| :heavy_check_mark: | a19edfc6 | [#30](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/30/) | 2021-08-10 13:30:48 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210810-132747-a19edf.zip) |
| :heavy_check_mark: | a19edfc6 | [#38](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/38/) | 2021-08-10 13:35:07 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210810-132747-a19edf.aar) |
| | | | | | | |
| :heavy_check_mark: | 195db56a | [#31](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/31/) | 2021-08-10 13:37:39 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210810-133507-195db5.zip) |
| :heavy_check_mark: | 195db56a | [#31](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/31/) | 2021-08-10 13:38:01 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210810-133502-195db5.zip) |
| :heavy_check_mark: | 195db56a | [#39](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/39/) | 2021-08-10 13:41:26 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210810-133514-195db5.aar) |
| | | | | | | |
| :heavy_check_mark: | 88826d8b | [#32](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/32/) | 2021-08-17 12:28:56 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210817-122606-88826d.zip) |
| :heavy_check_mark: | 88826d8b | [#32](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/32/) | 2021-08-17 12:29:29 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210817-122600-88826d.zip) |
| :heavy_check_mark: | 88826d8b | [#40](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/40/) | 2021-08-17 12:33:28 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210817-122600-88826d.aar) |
| | | | | | | |
| :heavy_check_mark: | c69c2658 | [#33](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/33/) | 2021-08-18 13:18:36 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210818-131559-c69c26.zip) |
| :heavy_check_mark: | c69c2658 | [#33](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/33/) | 2021-08-18 13:19:04 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210818-131554-c69c26.zip) |
| :heavy_check_mark: | c69c2658 | [#41](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/41/) | 2021-08-18 13:21:18 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210818-131554-c69c26.aar) |
| | | | | | | |
| :heavy_check_mark: | 0bf68dd9 | [#34](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/34/) | 2021-08-21 23:11:54 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210821-230944-0bf68d.zip) |
| :heavy_check_mark: | 0bf68dd9 | [#34](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/34/) | 2021-08-21 23:12:03 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210821-230944-0bf68d.zip) |
| :heavy_check_mark: | 0bf68dd9 | [#42](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/42/) | 2021-08-21 23:14:35 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210821-230939-0bf68d.aar) |
| | | | | | | |
| :heavy_check_mark: | 25aa6bd1 | [#35](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/35/) | 2021-08-27 12:48:34 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210827-124601-25aa6b.zip) |
| :heavy_check_mark: | 25aa6bd1 | [#35](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/35/) | 2021-08-27 12:49:01 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210827-124556-25aa6b.zip) |
| :heavy_check_mark: | 25aa6bd1 | [#43](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/43/) | 2021-08-27 12:51:55 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210827-124555-25aa6b.aar) |
| | | | | | | |
| :heavy_check_mark: | d94d44c1 | [#36](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/36/) | 2021-09-01 10:40:33 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210901-103800-d94d44.zip) |
| :heavy_check_mark: | d94d44c1 | [#36](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/36/) | 2021-09-01 10:40:52 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210901-103800-d94d44.zip) |
| :heavy_check_mark: | d94d44c1 | [#44](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/44/) | 2021-09-01 10:45:00 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210901-103800-d94d44.aar) |
| | | | | | | |
| :heavy_check_mark: | 09fbc165 | [#37](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/37/) | 2021-09-01 12:24:44 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210901-122247-09fbc1.zip) |
| :heavy_check_mark: | 09fbc165 | [#37](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/37/) | 2021-09-01 12:25:12 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210901-122252-09fbc1.zip) |
| :heavy_check_mark: | 09fbc165 | [#45](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/45/) | 2021-09-01 12:27:42 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210901-122247-09fbc1.aar) |
| | | | | | | |
| :heavy_check_mark: | 43c36576 | [#38](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/38/) | 2021-09-01 12:31:06 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210901-122912-43c365.zip) |
| :heavy_check_mark: | 43c36576 | [#38](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/38/) | 2021-09-01 12:31:21 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210901-122912-43c365.zip) |
| :heavy_check_mark: | 43c36576 | [#46](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/46/) | 2021-09-01 12:35:31 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210901-122912-43c365.aar) |
| | | | | | | |
| :heavy_check_mark: | f07b89ca | [#39](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/39/) | 2021-09-01 12:57:37 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210901-125552-f07b89.zip) |
| :heavy_check_mark: | f07b89ca | [#39](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/39/) | 2021-09-01 12:57:55 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210901-125552-f07b89.zip) |
| :heavy_check_mark: | f07b89ca | [#47](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/47/) | 2021-09-01 13:00:26 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210901-125552-f07b89.aar) |
| | | | | | | |
| :heavy_check_mark: | 82159329 | [#40](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/40/) | 2021-09-01 13:37:17 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210901-133448-821593.zip) |
| :heavy_check_mark: | 82159329 | [#40](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/40/) | 2021-09-01 13:37:18 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210901-133448-821593.zip) |
| :heavy_check_mark: | 82159329 | [#48](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/48/) | 2021-09-01 13:39:48 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210901-133448-821593.aar) |
| | | | | | | |
| :heavy_check_mark: | ff980f4e | [#41](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/41/) | 2021-09-08 13:18:45 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210908-131543-ff980f.zip) |
| :heavy_check_mark: | ff980f4e | [#41](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/41/) | 2021-09-08 13:19:17 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210908-131543-ff980f.zip) |
| :heavy_check_mark: | ff980f4e | [#49](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/49/) | 2021-09-08 13:23:38 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210908-131543-ff980f.aar) |
| | | | | | | |
| :heavy_check_mark: | 8774ed82 | [#42](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/42/) | 2021-09-14 11:15:34 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210914-111300-8774ed.zip) |
| :heavy_check_mark: | 8774ed82 | [#42](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/42/) | 2021-09-14 11:16:06 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210914-111257-8774ed.zip) |
| :heavy_check_mark: | 8774ed82 | [#50](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/50/) | 2021-09-14 11:19:32 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210914-111256-8774ed.aar) |
| | | | | | | |
| :x: | aac30bfb | [#51](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/51/) | 2021-09-21 15:17:19 | ~2 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/51/consoleText) |
| :x: | aac30bfb | [#43](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/43/) | 2021-09-21 15:17:23 | ~2 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/43/consoleText) |
| :x: | aac30bfb | [#43](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/43/) | 2021-09-21 15:17:52 | ~2 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/43/consoleText) |
| | | | | | | |
| :heavy_check_mark: | 0b6a7a46 | [#44](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/44/) | 2021-09-21 15:23:58 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210921-151745-0b6a7a.zip) |
| :heavy_check_mark: | 0b6a7a46 | [#44](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/44/) | 2021-09-21 15:26:06 | ~8 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210921-151755-0b6a7a.zip) |
| :heavy_check_mark: | 0b6a7a46 | [#52](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/52/) | 2021-09-21 15:29:33 | ~11 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210921-151745-0b6a7a.aar) |
| | | | | | | |
| :heavy_check_mark: | 4f4dc028 | [#45](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/45/) | 2021-09-21 22:10:30 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210921-220822-4f4dc0.zip) |
| :heavy_check_mark: | 4f4dc028 | [#45](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/45/) | 2021-09-21 22:10:44 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210921-220822-4f4dc0.zip) |
| :heavy_check_mark: | 4f4dc028 | [#53](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/53/) | 2021-09-21 22:13:40 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210921-220817-4f4dc0.aar) |
| | | | | | | |
| :heavy_check_mark: | 41651265 | [#46](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/46/) | 2021-09-21 22:18:16 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210921-221637-416512.zip) |
| :heavy_check_mark: | 41651265 | [#46](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/46/) | 2021-09-21 22:18:38 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210921-221642-416512.zip) |
| :heavy_check_mark: | 41651265 | [#54](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/54/) | 2021-09-21 22:20:58 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210921-221637-416512.aar) |
| | | | | | | |
| :heavy_check_mark: | 44658d4d | [#47](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/47/) | 2021-09-21 22:20:05 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210921-221819-44658d.zip) |
| :heavy_check_mark: | 44658d4d | [#47](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/47/) | 2021-09-21 22:21:04 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210921-221840-44658d.zip) |
| | | | | | | |
| :heavy_check_mark: | 851f5c0f | [#48](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/48/) | 2021-09-21 22:22:36 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210921-222009-851f5c.zip) |
| :heavy_check_mark: | 851f5c0f | [#48](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/48/) | 2021-09-21 22:23:17 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210921-222108-851f5c.zip) |
| :heavy_check_mark: | 851f5c0f | [#55](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/55/) | 2021-09-21 22:25:37 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210921-222100-851f5c.aar) |
| | | | | | | |
| :heavy_check_mark: | 2244c916 | [#49](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/49/) | 2021-09-21 22:33:12 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210921-223117-2244c9.zip) |
| :heavy_check_mark: | 2244c916 | [#56](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/56/) | 2021-09-21 22:36:27 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210921-223117-2244c9.aar) |
| :heavy_check_mark: | 2244c916 | [#49](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/49/) | 2021-09-21 22:36:37 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210921-223117-2244c9.zip) |
| | | | | | | |
| :heavy_check_mark: | ed4b5abe | [#50](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/50/) | 2021-09-21 22:36:24 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210921-223315-ed4b5a.zip) |
| | | | | | | |
| :heavy_check_mark: | 0f78952f | [#51](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/51/) | 2021-09-21 22:38:16 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210921-223627-0f7895.zip) |
| :heavy_check_mark: | 0f78952f | [#50](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/50/) | 2021-09-21 22:39:25 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210921-223639-0f7895.zip) |
| :heavy_check_mark: | 0f78952f | [#57](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/57/) | 2021-09-21 22:40:34 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210921-223629-0f7895.aar) |
| | | | | | | |
| :heavy_check_mark: | 02ea1afc | [#51](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/51/) | 2021-09-22 09:36:52 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210922-093436-02ea1a.zip) |
| :heavy_check_mark: | 02ea1afc | [#52](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/52/) | 2021-09-22 09:38:58 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210922-093436-02ea1a.zip) |
| :heavy_check_mark: | 02ea1afc | [#58](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/58/) | 2021-09-22 09:39:25 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210922-093436-02ea1a.aar) |
| | | | | | | |
| :heavy_check_mark: | ed3d3767 | [#53](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/53/) | 2021-09-24 21:28:00 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210924-212523-ed3d37.zip) |
| :heavy_check_mark: | ed3d3767 | [#52](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/52/) | 2021-09-24 21:28:00 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210924-212523-ed3d37.zip) |
| :heavy_check_mark: | ed3d3767 | [#59](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/59/) | 2021-09-24 21:30:24 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210924-212523-ed3d37.aar) |
| | | | | | | |
| :heavy_check_mark: | 01d97d6b | [#53](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/53/) | 2021-09-24 21:29:43 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210924-212805-01d97d.zip) |
| :heavy_check_mark: | 01d97d6b | [#54](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/54/) | 2021-09-24 21:29:45 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210924-212802-01d97d.zip) |
| :heavy_check_mark: | 01d97d6b | [#60](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/60/) | 2021-09-24 21:34:17 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210924-213027-01d97d.aar) |
| | | | | | | |
| :heavy_check_mark: | 47e8f4bc | [#54](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/54/) | 2021-09-24 21:32:21 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210924-213048-47e8f4.zip) |
| :heavy_check_mark: | 47e8f4bc | [#55](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/55/) | 2021-09-24 21:32:40 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210924-213048-47e8f4.zip) |
| :heavy_check_mark: | 47e8f4bc | [#61](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/61/) | 2021-09-24 21:38:46 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210924-213420-47e8f4.aar) |
| | | | | | | |
| :heavy_check_mark: | e1bf4449 | [#55](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/55/) | 2021-09-27 11:18:24 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210927-111620-e1bf44.zip) |
| :heavy_check_mark: | e1bf4449 | [#56](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/56/) | 2021-09-27 11:20:07 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210927-111620-e1bf44.zip) |
| :heavy_check_mark: | e1bf4449 | [#62](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/62/) | 2021-09-27 11:23:41 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210927-111620-e1bf44.aar) |
| | | | | | | |
| :heavy_check_mark: | 325ced0d | [#57](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/57/) | 2021-09-29 08:24:07 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210929-082148-325ced.zip) |
| :heavy_check_mark: | 325ced0d | [#56](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/56/) | 2021-09-29 08:24:19 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210929-082148-325ced.zip) |
| :heavy_check_mark: | 325ced0d | [#63](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/63/) | 2021-09-29 08:28:32 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210929-082148-325ced.aar) |
| | | | | | | |
| :heavy_check_mark: | 181e4947 | [#57](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/57/) | 2021-09-29 08:36:34 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210929-083409-181e49-pr2151.zip) |
| :heavy_check_mark: | 181e4947 | [#58](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/58/) | 2021-09-29 08:38:05 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210929-083409-181e49-pr2151.zip) |
| :heavy_check_mark: | 181e4947 | [#64](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/64/) | 2021-09-29 08:39:04 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210929-083404-181e49-pr2151.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | d25b2813 | [#58](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/58/) | 2021-09-29 14:57:21 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210929-145409-d25b28-pr2151.zip) |
| :heavy_check_mark: | d25b2813 | [#59](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/59/) | 2021-09-29 15:01:04 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210929-145404-d25b28-pr2151.zip) |
| :heavy_check_mark: | d25b2813 | [#65](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/65/) | 2021-09-29 15:01:58 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210929-145404-d25b28-pr2151.aar) |
| | | | | | | |
| :heavy_check_mark: | a73aaabe | [#60](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2151/60/) | 2021-09-30 11:52:56 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210930-115039-a73aaa-pr2151.zip) |
| :heavy_check_mark: | a73aaabe | [#59](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2151/59/) | 2021-09-30 11:53:09 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210930-115039-a73aaa-pr2151.zip) |
| :heavy_check_mark: | a73aaabe | [#66](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2151/66/) | 2021-09-30 11:55:39 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210930-115039-a73aaa-pr2151.aar) |

Samyoul committed:
Added new Settings field ProfilePicturesShowTo

Samyoul committed:
Updated contact key gen in test so index call wouldn't be ambiguous

Samyoul committed:
Updated tests to include tests on IdentityImage.Encrypted state

Samyoul committed:
Refactor so both Messenger and MessageHandler could use code from a shsared source

Samyoul committed:
Updated tests to match refactor

Samyoul committed:
Attempting to resolved 'no such column: profile_pictures_show_to'

Have resolved the settings table not found but now the column is not migrating for some reason

cammellos committed:
rename file

Samyoul committed:
Tests and linting

cammellos reviewed (approved):
Looks good to me, the only question I have, what happens to old client if they receive an encrypted image?
If they drop it/don't display it, it's all fine, if they show some garbled picture then we might want to use a separate field for the encrypted image in protobuf.

Samyoul commented:
> Looks good to me, the only question I have, what happens to old client if they receive an encrypted image?

Excellent point. So the image payload is passed by the `HandleChatIdentity` func to `images.GetPayloadDataURI()` and would almost 100% get a `image format not supported` error which would trigger a `logger.Warn("failed to handle ChatIdentity", zap.Error(err))` and set `allMessagesProcessed` to false. Which I think is safe behaviour.

---

Although that would trigger for any image payload that is encrypted on the `ChatIdentity`. At the moment a `ChatIdentity` only has a total of 2 potential image payloads (`thumbnail` and `large`), but in the future this may not be the case (some maybe encrypted and other not ... maybe).

I've also realised that a single image parsing error would cause all of the attached images to be rejected, I've added a `TODO` to this functionality I'll address it in another PR.

Also I've realised that a single `IdentityImage` parse failure would result in the entire `ChatIdentity` parse to fail, that needs addressing also.

Samyoul commented:
Awaiting a successful manual test of https://github.com/status-im/status-react/pull/11768 before merge of this PR

ghost commented:
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

Samyoul commented:
Failing Scenarios

```
{
"chat context type": "private-chat", 
"profile picture Show Settings": "ShowToEveryone", 
"profile picture View Settings": "ViewFromContactsOnly", 
"bob in Alice's Contacts": false,
"alice in Bob's Contacts": true
}
expect to receive a profile pic? {"result": true}
```

[Full Logs](https://gist.github.com/Samyoul/9bcebe7858ba9ae139811a62eb901d85)

---

```
{
"chat context type": "private-chat", 
"profile picture Show Settings": "ShowToEveryone", 
"profile picture View Settings": "ViewFromEveryone", 
"bob in Alice's Contacts": true, 
"alice in Bob's Contacts": false
}
expect to receive a profile pic?{"result": true}
```

[Full Logs](https://gist.github.com/Samyoul/b35430936277f4fbb1872a7485919b84)

---

```
{
"chat context type": "private-chat", 
"profile picture Show Settings": "ShowToContactsOnly", 
"profile picture View Settings": "ViewFromEveryone", 
"bob in Alice's Contacts": true, 
"alice in Bob's Contacts": false
}
expect to receive a profile pic?{"result": true}
```

[Full Logs](https://gist.github.com/Samyoul/21ac0f56afe40e0ab7856b4f265fb45f)

---


```
{
"chat context type": "private-chat", 
"profile picture Show Settings": "ShowToContactsOnly", 
"profile picture View Settings": "ViewFromContactsOnly", 
"bob in Alice's Contacts": true,
"alice in Bob's Contacts": true
}
expect to receive a profile pic?{"result": true}
```

[Full Logs](https://gist.github.com/Samyoul/7fd78c45f9fa3b5e72a299de8a33d0c6)

---

cammellos commented on protocol/messenger.go:
```diff
@@ -600,14 +608,24 @@ func (m *Messenger) publishContactCode() error {
 		m.logger.Error("could not build contact code advertisement", zap.Error(err))
 	}
 
+	m.logger.Debug("initing cca if not set")
```
You might want to remove this logs

Samyoul replied:
Ah, I thought I'd got them all.

cammellos reviewed (approved):
looks good, just some extra logs

cammellos committed:
Resend on new contact
